common/domainHash.js

```diff
@@ -0,0 +1,49 @@
+function byteToHex(byte) {
+  return ('0' + byte.toString(16)).slice(-2);
+}
+
+function generateId(len = 32) {
+  var arr = new Uint8Array(len / 2);
+  window.crypto.getRandomValues(arr);
+  return Array.from(arr, byteToHex).join("");
+}
+
+function gen_random32() {
```

Comment:
AFAIK, this function is a part of the helpers.js, what is the reason of not using the one in helpers.js?